test/interface/ad.jl

```diff
@@ -18,11 +22,19 @@ h(p, t) = [p[4]]
     dot(diff, diff)
   end
 
+  # with delay length estimation
   p = [1.0, 1.0, 1.0, 0.0, 1.0]
-  findiff = Calculus.finite_difference(test, p)
+  findiff = FiniteDiff.finite_difference_gradient(test, p)
   fordiff = @test_logs (:warn, r"^dt <= dtmin") ForwardDiff.gradient(test, p)
-
   @test_broken findiff ≈ fordiff
+
+  # without delay length estimation
+  p = [1.0, 1.0, 0.0, 1.0]
+  findiff2 = FiniteDiff.finite_difference_gradient(p -> test(vcat(1, p)), p)
```

Comment:
we need to do lower tolerances?

Reply:
Lower tolerances for the DDE solver? We test with `abstol = 1e-14` and `reltol = 1e-14`.

BTW finite and forward diff yield really completely different results, for the gradient example we get
```julia
(findiff, findiff2) = ([403302.77650387725, 3.169348882494423e6, 408139.9972998477, -3.289280466135436e6, 2.8082672710581566e6], [3.169348882494423e6, 408139.9972998477, -3.289280466135436e6, 2.8082672710581566e6])
(fordiff, fordiff2) = ([0.0, -1853.468233232502, 0.0, 1275.361852383486, -2705.591974038044], [-1853.468233232502, 0.0, 1275.361852383486, -2705.591974038044])
```